main.js

```diff
@@ -0,0 +1,802 @@
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'Enter',
+    },
+  ],
+  [
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'ShiftLeft',
+    },
+    {
+      value1: 'z',
+      value2: 'я',
+      alt1: null,
+      alt2: null,
+      code: 'KeyZ',
+    },
+    {
+      value1: 'x',
+      value2: 'ч',
+      alt1: null,
+      alt2: null,
+      code: 'KeyX',
+    },
+    {
+      value1: 'c',
+      value2: 'с',
+      alt1: null,
+      alt2: null,
+      code: 'KeyC',
+    },
+    {
+      value1: 'v',
+      value2: 'м',
+      alt1: null,
+      alt2: null,
+      code: 'KeyV',
+    },
+    {
+      value1: 'b',
+      value2: 'и',
+      alt1: null,
+      alt2: null,
+      code: 'KeyB',
+    },
+    {
+      value1: 'n',
+      value2: 'т',
+      alt1: null,
+      alt2: null,
+      code: 'KeyN',
+    },
+    {
+      value1: 'm',
+      value2: 'ь',
+      alt1: null,
+      alt2: null,
+      code: 'KeyM',
+    },
+    {
+      value1: ',',
+      value2: 'б',
+      alt1: '<',
+      alt2: null,
+      code: 'Comma',
+    },
+    {
+      value1: '.',
+      value2: 'ю',
+      alt1: '>',
+      alt2: null,
+      code: 'Period',
+    },
+    {
+      value1: '/',
+      value2: '.',
+      alt1: '?',
+      alt2: ',',
+      code: 'Slash',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'ArrowUp',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'ShiftRight',
+    },
+  ],
+  [
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'ControlLeft',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'MetaLeft',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'AltLeft',
+    },
+    {
+      value1: ' ',
+      value2: ' ',
+      alt1: null,
+      alt2: null,
+      code: 'Space',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'AltRight',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'ControlRight',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'ArrowLeft',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'ArrowDown',
+    },
+    {
+      value1: '',
+      value2: '',
+      alt1: null,
+      alt2: null,
+      code: 'ArrowRight',
+    },
+  ],
+];
+
+const keyboard = {
+  elements: {
+    main: null,
+    keys: [],
+  },
+
+  properties: {
+    value: '',
+    capslock: false,
+    shift: false,
+    lang: localStorage.getItem('lang') || 'eng',
+    lastCaretPos: 0,
+  },
+
+  initilize() {
+    localStorage.setItem('isRus', this.properties.isRus);
```

Comment:
Initial value is always undefined, maybe, need to check initial value and set true or false.